### PR TITLE
Avoid overly-general rescue in callback controller

### LINKED ISSUE #1527: Avoid overly-general `rescue` of errors in `CallbackController`
In `CallbackController`, we [rescue from _any_ kind of `StandardError`.](https://github.com/Shopify/shopify_app/blob/6f8080da5e765859091ec133dff0370999917cc1/app/controllers/shopify_app/callback_controller.rb#L20-L22) This can mask coding errors such as calling an undefined method, which may later result in an infinite redirect or other problem.

Instead, we should only rescue from error swhich are _expected_ as part of the OAuth flow, and allow other kinds of errors to bubble up.

### PR DESCRIPTION
**Pull request by andyw8**
Closes #1527

~Still to test, just looking for feedback on the approach.~

### CODE REVIEW

File: test/controllers/callback_controller_test.rb [hunk 1]
```diff
@@ -35,13 +35,13 @@ class CallbackControllerTest < ActionController::TestCase
     end
 
     test "#callback flashes error when omniauth is not present" do
-      get :callback, params: { shop: "shop" }
```
Comment (andyw8) [inline test/controllers/callback_controller_test.rb:38]: The `rescue` was masking these failing tests.

**Comment by andyw8** [comment]
I plan to change this to emit a deprecation warning instead of raising.

**Review by nelsonwittwer — APPROVED** [review]

**Comment by andyw8** [comment]
@nelsonwittwer ready for a final review.

**Pull request merged**